lib/x509_wrapper.cpp

```diff
@@ -111,6 +116,30 @@ void X509Wrapper::update_validity() {
                sec; // Convert days to seconds
 }
 
+bool X509Wrapper::is_child(const X509Wrapper& parent) const {
+    if ((this == &parent) || (*this == parent)) {
+        return false;
+    }
```

Comment:
Wouldn't a self-signed certificate be its own parent?

Reply:
Feedback from @AssemblyJohn : A certificate is never its own child, even when self-signed. This makes building certificate hierarchies cleaner.